scripts/src/modules/insights/accordion_functionality/set_heading_positions.js

```diff
@@ -0,0 +1,29 @@
+import convert_px_property_to_number from "./convert_px_property_to_number";
+
+// This function calculates the position of each heading. In order to prevent expanded sections from causing the heading 
+// position being calculated too far down the page, the top position of the first heading is taken as the first index. 
+// The height of subsequent headings is calculated by adding its height to the previous index.
+export default function set_heading_positions(sectionHeadings) {
+    const positions = [];
+
+    sectionHeadings.each(function(index) {
+        if($(this).attr("id") === $(sectionHeadings[0]).attr("id")) {
+            positions.push($(this).offset().top);
+        }
+        else {
+            const prevHeading = sectionHeadings[index - 1];
+
+            // This accounts for any padding, margin, and borders that have been applied to the heading in order to correctly
+            // calculate its height.
+            const sectionHeadingPadding = convert_px_property_to_number($(prevHeading).css("padding-top")) + convert_px_property_to_number($(prevHeading).css("padding-bottom"));
```

Comment:
Could you use jQuery's outerHeight here? It looks like that would calculate the height with padding, border, and margin for you:

https://www.w3schools.com/jquery/html_outerheight.asp

https://api.jquery.com/outerheight/